static/js/settings_org.js

```diff
@@ -860,28 +865,21 @@ export function register_save_discard_widget_handlers(
         switch (subsection) {
             case "msg_editing": {
                 const edit_limit_setting_value = $("#id_realm_msg_edit_limit_setting").val();
+                data.allow_message_editing = $("#id_realm_allow_message_editing").prop("checked");
                 switch (edit_limit_setting_value) {
-                    case "never": {
-                        data.allow_message_editing = false;
-
-                        break;
-                    }
                     case "custom_limit": {
                         data.message_content_edit_limit_seconds = parse_time_limit(
                             $("#id_realm_message_content_edit_limit_minutes"),
                         );
-                        data.allow_message_editing = true;
 
                         break;
                     }
                     case "any_time": {
-                        data.allow_message_editing = true;
                         data.message_content_edit_limit_seconds = JSON.stringify("unlimited");
 
                         break;
                     }
                     default: {
-                        data.allow_message_editing = true;
```

Comment:
I think now we can also remove this and send only changed values the API and not all settings in subsection, as we no longer control multiple settings using single UI element. Though some settings are disabled according to others but that is the UI part, and we need not send all the settings to the API.

Reply:
That sounds correct.